app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java

```diff
@@ -389,6 +403,45 @@ private int getClickedGraphicId(float screenX, float screenY) {
         private boolean isEntityClicked(int graphicId) {
             return graphicId != -1;
         }
+
+        private void handleKmlEntityClicks(float screenX, float screenY) {
+            KmlEntityInfo info = getClickedKmlInfo(screenX, screenY);
+            if (info != null) {
+                notifyKmlEntityClicked(info);
+            }
+        }
+
+        private KmlEntityInfo getClickedKmlInfo(float screenX, float screenY) {
+            Set<String> layerIds = mVectorLayerIdToKmlLayerMap.keySet();
+            for (String id : layerIds) {
+                KmlLayer layer = mVectorLayerIdToKmlLayerMap.get(id);
+                KmlNode[] kmlNodes = getKmlNodes(screenX, screenY, layer);
+                if (kmlNodes.length > 0) {
+                    return createKmlEntityInfo(kmlNodes[0], id);
+                }
+            }
+            return null;
+        }
+
+        private KmlNode[] getKmlNodes(float screenX, float screenY, KmlLayer layer) {
+            return layer.getKmlNodes(screenX, screenY, Constants
+                    .VIEWER_ENTITY_CLICKING_TOLERANCE_DP, false);
+        }
+
+        private KmlEntityInfo createKmlEntityInfo(KmlNode kmlNode, String layerId) {
+            String entityName = kmlNode.getName();
+            String description = null;
+            if (kmlNode.getType() == KmlNode.Type.PLACEMARK) {
```

Comment:
Try to extract a description for all types of KMLs.
(Make sure to check for nulls when extracting so we don't get NPE on various types of data - log description extraction failures)